internal/vercheck/vercheck.go

```diff
@@ -62,14 +71,84 @@ func SelfUpdate(stdOut, stdErr io.Writer) error {
 		return errors.WithStack(err)
 	}
 
-	fmt.Fprint(stdErr, "Latest version: ")
+	fmt.Fprint(stdOut, "Latest version: ")
+	return triggerUpdateAndPrintNewVersion(stdOut, stdErr)
+}
+
+// selfUpdateBinary will update the binary to the latest version.
+func selfUpdateBinary(stdOut, stdErr io.Writer) error {
+	resp, err := http.Get(fmt.Sprintf("https://releases.jetpack.io/devbox/%s/version", currentChannel))
+	if err != nil {
+		fmt.Fprint(
+			stdErr,
+			"Failed to get latest version. "+
+				"Please try again, or run the command from https://www.jetpack.io/devbox/docs/installing_devbox/",
+		)
+		return errors.WithStack(err)
+	}
+	defer resp.Body.Close()
+
+	newVersion, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	// Remove the version cache file, so that when we re-create it the expiration
+	// time is reset.
+	_ = os.Remove(versionCacheFilePath())
+
+	// Write the new version to the version cache file.
+	if err := os.WriteFile(versionCacheFilePath(), newVersion, 0644); err != nil {
+		return errors.WithStack(err)
+	}
+
+	return triggerUpdateAndPrintNewVersion(stdOut, stdErr)
+}
+
+func triggerUpdateAndPrintNewVersion(stdOut, stdErr io.Writer) error {
+	fmt.Fprintln(stdOut, "triggering update...")
+
 	exe, err := os.Executable()
 	if err != nil {
 		return errors.WithStack(err)
 	}
-	cmd = exec.Command(exe, "version")
-	// The output of version is incidental, so just send it all to stdErr
-	cmd.Stdout = stdErr
```

Comment:
I'm a bit confused about 

> I feel this is better so user is informed about the new version.

Both stdout and stderr get printed to terminal by default so anything we want to warn the user about can go in stderr. The main heuristic I use is:

* If the command explicitly asking for output e.g. `devbox global ls` or `devbox shellenv`. The output is the _reason_ for these commands. From a user perspective, they are not meant to perform actions as much as simply return/print text.
* If the command meant to do some action "x" and then print the status or information associated to the performance of this command, in that case I send to stderr. 

Anyway, we don't strictly follow this and since update requires a tti anyway (it asks a question and does sudo) it's not a big deal so not a blocker. Just sharing my process.

Reply:
Just to clarify, stderr is visible to the user. I do want users to see diagnostic info especially errors that occur during normal operation. 

So `devbox version update` will show the latest version and `devbox add <pkg>` will show success (but to stderr).